### ISSUE TITLE
Linking fails on target i686-pc-windows-gnu because of unrecognized output object file format

### ISSUE THREAD

**Issue by utoalex** [nabijaczleweli/rust-embed-resource#34]
Trying to compile anything depending on a resource (`.rc`) file using on an amd64 computer using target `i686-pc-windows-gnu` yields the following error:

```
linking with `i686-w64-mingw32-gcc` failed: exit code: 1
  |
  = note: "i686-w64-mingw32-gcc" <lots of args...>
  = note: <project root>\target\debug\build\<project name>-<hash>\out/libresources.a: file not recognized: File format not recognized 
```

This error seems to stem from an incorrect use of `mingw-w64`'s `windres.exe` resource compiler (just in this case), in [`windows_not_msvc.rs`](https://github.com/nabijaczleweli/rust-embed-resource/blob/c0855cc6a48f0af18e555a0ce15d9964fa5e1dbe/src/windows_not_msvc.rs#L22).

By default,`mingw-w64`'s `windres.exe` compiles the input resource file into an `amd64` object file format (at least when running on an amd64 computer), which is incompatible with `i686` targets, thereby causing linking to fail.
A fix for this would consist in detecting that `i686` machines are being targeted (using `cfg!(target_arch = "x86_64")`), and using flag `--target=pe-i386` just after `-O coff`/`--output-format=coff`, like this:

```
windres --output-format=coff --target=pe-i386 <resource file>
```

As a temporary workaround, using the above command line manually to compile the resource file and replacing the invalid `libresources.a` resource file generated by this crate with our manually compiled file fixes builds broken by this issue.

**Comment by nabijaczleweli** [comment]
Can you try the current `master` branch (at least ae7c38bbea8a999b6d948bdf58e36cd37f067a16)?

**Comment by utoalex** [comment]
Just tried it and it works fine now, thanks!

**Comment by nabijaczleweli** [comment]
Hog. Released in [`v1.6.2`](https://github.com/nabijaczleweli/rust-embed-resource/releases/tag/v1.6.2), thanks for the draft of the fix!
